numpyro/distributions/discrete.py

```diff
@@ -91,7 +91,8 @@ def sample(self, key, sample_shape=()):
 
     @validate_sample
     def log_prob(self, value):
-        return xlogy(value, self.probs) + xlog1py(1 - value, -self.probs)
+        ps_clamped = clamp_probs(self.probs)
```

Comment:
@julianstastny sorry, I forgot to follow up. To make the test pass (and to avoid issues when users provide an integer `probs=0`, could you help me change [this line](https://github.com/pyro-ppl/numpyro/blob/507acedba02e1b47f3bc4d0d7d8ade129bb2516b/numpyro/distributions/util.py#L398) to
```
finfo = jnp.finfo(jnp.result_type(probs, float))
```
Thanks!

Reply:
Done!